janus_server/src/aggregator.rs

```diff
@@ -0,0 +1,76 @@
+//! Common functionality for PPM aggregators
+use crate::{
+    hpke::{HpkeRecipient, Label},
+    message::{Role, TaskId},
+};
+use http::{header::CACHE_CONTROL, StatusCode};
+use prio::codec::Encode;
+use std::{future::Future, net::SocketAddr};
+use warp::{filters::BoxedFilter, reply, trace, Filter, Reply};
+
+/// Constructs a Warp filter with an aggregator's endpoints.
+fn aggregator_filter(task_id: TaskId) -> BoxedFilter<(impl Reply,)> {
+    let hpke_recipient =
+        HpkeRecipient::generate(task_id, Label::InputShare, Role::Client, Role::Leader);
+
+    let hpke_config_encoded = hpke_recipient.config.get_encoded();
+
+    warp::get()
+        .and(warp::path("hpke_config"))
```

Comment:
nit: I think it would be best to chain the `warp::path` filter ahead of `warp::get`. It seems to be the idiomatic order in examples. Plus each instance `warp::get`, `warp::post` and friends that appears first in our top-level router will emit WARN level log messages about 405 Method Not Allowed for each request that does not pass the filter.